### DISCUSSION TITLE
Access collection from @request parameter in Rules

### DISCUSSION THREAD

**Discussion by iskaa02** [pocketbase/pocketbase#774] [Q&A]
i have relation 'a' that have relation 'b' in one of it's fields
in the create rule trying to do `a.b.name=something` didn't work because the fields are empty, and on update rule i set data manually where the condition is true then trying to do that will work for the first time when the condition is true, then it will stop working.

what i thought about is the ability to use something like `a(@request.data.a_id).b.name`, Or having a `new` field in the create and update rule where `new` have the fields before inserting or updating, for example `new.a.b.name=something`

**Comment by ganigeorgiev** [comment]
> in the create rule trying to do a.b.name=something didn't work because the fields are empty

I've just tested it and it should work. The create rule is applied after a transactional "dry save" of the record, so you should have access to the relation data fields. Or do you mean something else?

**Reply by ganigeorgiev** [reply]
For update, you don't have direct access to the newly submitted relation data fields, but you can combine it with  a `@collection.*` filter.

I'll have to think a little more on this but eventually we may add support for direct nested relation fields access from the `@request.data.*` filter.

**Reply by ganigeorgiev** [reply]
@iskaa02 _Just to let you know that this is now supported with the latest [v0.11.0 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.11.0)._
